azavea/franklin
# View collection item footprints as MVT layer

## Overview

This PR:

- upgrades Postgres in the dev container to 12.2 (+ postgis 3)
- adds a tile info endpoint to collections 
- adds a tiles link to collections
- renders collection item footprints for collections as an MVT layer

### Checklist

- ~New tests have been added or existing tests have been modified~

### Notes

It's difficult to see the mvt layer whole. I'm thinking about whether there's an easy way to adapt Kenny's or Alex's bakeoff projects for it, since they both use mapbox gl js, and as far as I can tell I'd just (:tm:) have to change the source to get a different layer, either [here](https://github.com/alkamin/tsm-bakeoff-ts/blob/8a935acc373b630444666ba0c56b726cee251941/src/components/SchoolMap.tsx#L44-L47) or [here](https://github.com/kshepard/tsm-bakeoff-elm/blob/master/src/index.js#L19-L36).

### Testing Instructions

- remove the database container (the updated postgres image can't use the old PG_DATA)
- `./scripts/update`
- import the berlin data
- start the server `--with-tiles` and `--with-transactions`
- add a bunch of items to the berlin collection -- you can use the attached python script to do so

[item-maker.zip](https://github.com/azavea/franklin/files/4525313/item-maker.zip)

- navigate the `http://localhost:9090/collections/berlin`
- follow its tile link
- fill in its tile link with `WebMercatorQuad` for the matrix set and `13/4407/2682` (it's a tile matching part of Berlin).
- `http 'http://localhost:9090/tiles/collections/berlin/footprint/WebMercatorQuad/13/4407/2682' > mvt` -- it should succeed surprisingly fast
- install [`mvtview`](https://github.com/mapbox/mvtview)
- `mvtview mvt` -- you should see a bunch of chaotic lines for the footprints of the bogus items you created in a Polygon layer

Closes #196
